### PR TITLE
Milestone #1: Setup and Controllers

### PR DESCRIPTION
**Pull request by Anjacodes**
In this milestone, I:

* Created a new app and initialized it with Git.
* Made sure that the project has Postgres database set up.
* Made sure that the app handles the URLs used in the wireframes.
* For each URL I created:
        * A route.
        * An action in the correct controller.
        * A view file.
* View files include only placeholders written in plain HTML.
* Made sure I added code to handle only the 4 required URLs.
* Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
* Made sure that to use correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
* Made sure to document my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)[](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project.md#ruby-requirements).
* Followed the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -2,5 +2,13 @@
...
-  # root "articles#index"
+  root "users#index"
+
+  get "/users", to: "users#index"
+  get "/users/:id", to: "users#show"
+  get "/users/:id/posts", to: "posts#index"
+  get "/users/:id/posts/3", to: "posts#show"
+  # resources :users do
+  #   resources :posts
+  # end
```
Comment (Dipeshtwis) [inline config/routes.rb]: - [ ] I would recommend maintaining a RESTful URL and also the correct URL.

`get "/users/:id/posts/3", to: "posts#show"` :- this is kind of specific URL which only go to the post-show page whose post_id is 3. You can use what I have mentioned below.

```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

File: db/migrate/20220322165424_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :body
+      t.references :user, null: false, foreign_key: true
+
+      t.timestamps
+    end
+  end
+end
```
Comment (Dipeshtwis) [inline db/migrate/20220322165424_create_posts.rb]: - The post and user migrations or making associations were not in this milestone. Thanks for the extra work but always work according to your milestone need. :+1:

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Anjacodes, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220322165424_create_posts.rb (hunk 2 above)
Comment (Anjacodes) [reply db/migrate/20220322165424_create_posts.rb]: Hi @Dipeshtwis, I was having my doubts about this but decided to give it a try 😆 I have now deleted the migrations and altered the HTML files to only display dummy text.

File: config/routes.rb (hunk 1 above)
Comment (Anjacodes) [reply config/routes.rb]: Thank you for this awesome tip @Dipeshtwis 🙏 . I was wondering how to do the nesting here without using resources but this is so much better. I have gone ahead and implemented that!

**Review by DeliceLydia — APPROVED** [review]
## Hi Anja, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**